packages/legend-query-builder/src/stores/QueryBuilderResultState.ts

```diff
@@ -323,13 +343,27 @@ export class QueryBuilderResultState {
 
       stopWatch.record();
       try {
-        this.executionPlanState.setRawPlan(rawPlan);
+        if (!options?.hideExecutionPlanViewer) {
```

Comment:
This function is a bit messy. 
Could we keep it just to generating a rawplan. 
I think we can add a type here 
`ExecutionPlanResult`
that holds 
{
   execPlan: RawExecutionPlan,
debugText?: string| undefined.
}

Then the various query actions used this as needed.